### ISSUE TITLE
Missing stubs for Schema\ColumnDefinition fluent methods (nullable, default, after, etc.)

### ISSUE THREAD

**Issue by alies-dev** [psalm/psalm-plugin-laravel#496]
## Description

`Illuminate\Database\Schema\ColumnDefinition` uses `__call()` to provide fluent methods like `->nullable()`, `->default()`, `->after()`, `->index()`, `->unique()`, `->primary()`, `->charset()`, `->storedAs()`, and `->change()`. The plugin does not provide stubs for these methods, causing `UndefinedMagicMethod` errors at Psalm errorLevel 1.

Similarly affected:
- `Illuminate\Database\Schema\ForeignIdColumnDefinition::nullable()` (28 occurrences)
- `Illuminate\Database\Schema\ForeignKeyDefinition::references()` (7 occurrences)

## Impact

**~359 false positives** across migration files in [monicahq/monica](https://github.com/monicahq/monica) (262 project files). This affects **every Laravel application** that uses standard migration patterns, making errorLevel 1 unusable for `database/migrations/`.

Breakdown:
- `ColumnDefinition::nullable` — 196
- `ColumnDefinition::default` — 57
- `ForeignIdColumnDefinition::nullable` — 28
- `ColumnDefinition::after` — 17
- `ForeignKeyDefinition::references` — 7
- `ColumnDefinition::index` — 6
- `ColumnDefinition::primary` — 5
- Other (`unique`, `charset`, `storedAs`, `change`) — ~43

## Reproduction

Found while auditing [monicahq/monica](https://github.com/monicahq/monica) at Psalm errorLevel 1.

**File:** `database/migrations/2013_04_25_132851_create_accounts_table.php:17`

```php
Schema::create('accounts', function (Blueprint $table) {
    $table->uuid('id');
    $table->primary('id');
    $table->integer('storage_limit_in_mb')->default(0); // UndefinedMagicMethod
    $table->timestamps();
});
```

**File:** `database/migrations/2014_10_12_000010_create_vaults_table.php:23`

```php
$table->string('description')->nullable(); // UndefinedMagicMethod
$table->foreignIdFor(Template::class, 'default_template_id')
    ->nullable()       // UndefinedMagicMethod on ForeignIdColumnDefinition
    ->constrained('templates')
    ->nullOnDelete();
```

## Expected Behavior

Psalm should recognize fluent methods on `ColumnDefinition`, `ForeignIdColumnDefinition`, and `ForeignKeyDefinition` without errors.

## Proposed Fix

Add a stub file `stubs/common/Illuminate/Database/Schema/ColumnDefinition.stubphp` declaring the fluent methods. The full list is defined in `ColumnDefinition`'s docblock `@method` annotations in Laravel source. Key methods:

```php
namespace Illuminate\Database\Schema;

class ColumnDefinition
{
    public function nullable(bool $value = true): static {}
    public function default(mixed $value): static {}
    public function after(string $column): static {}
    public function index(?string $indexName = null): static {}
    public function unique(?string $indexName = null): static {}
    public function primary(): static {}
    public function charset(string $charset): static {}
    public function change(): static {}
    // ... etc from Laravel's @method annotations
}
```

Similar stubs needed for `ForeignIdColumnDefinition` and `ForeignKeyDefinition`.
